server.js

```diff
@@ -3,27 +3,107 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
+import Thought from './models/thought'
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/hanna-happyThoughts" 
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
+const endpoints = require('express-list-endpoints')
+
+const POST_LIKE_ERROR = 'Error: could not add like on message with id:'
+const GET_ENDPOINTS_ERROR = 'Error: No endpoints found'
 
-// Add middlewares to enable cors and json body parsing
+// MIDDLEWARES (to enable cors and json body parsing)
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
+// RESET DATABASE
+if (process.env.RESET_DATABASE) {
+  const seedDatabase = async () => {
+    await Thought.deleteMany()
+  }
+  seedDatabase()
+}
+
+//////ROUTES (ENDPOINTS)////
+
+//GET: All available endpoints with express-list-endpoints
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  if (res) {
+    res.status(200).send(endpoints(app));
```

Comment:
I think it's really good that you are using the list-endpoints library, so nice to get an overview of all the available endpoints.